tests/Umbraco.Tests.Integration/Umbraco.Infrastructure/Services/EntityServiceTests.cs

```diff
@@ -931,6 +931,95 @@ public void EntityService_GetPathKeys_ReturnsExpectedKeys()
 
     }
 
+    [Test]
+    public void EntityService_Siblings_ReturnsExpectedSiblings()
+    {
+        var children = CreateSiblingsTestData();
+
+        var taget = children[1];
+
+        var result = EntityService.GetSiblings(taget.Key, UmbracoObjectTypes.Document, 1, 1).ToArray();
+        Assert.AreEqual(3, result.Length);
+        Assert.IsTrue(result[0].Key == children[0].Key);
+        Assert.IsTrue(result[1].Key == children[1].Key);
+        Assert.IsTrue(result[2].Key == children[2].Key);
+    }
+
+    [Test]
+    public void EntityService_Siblings_SkipsTrashedEntities()
+    {
+        var children = CreateSiblingsTestData();
+
+        var trash = children[1];
+        ContentService.MoveToRecycleBin(trash);
+
+        var taget = children[2];
+        var result = EntityService.GetSiblings(taget.Key, UmbracoObjectTypes.Document, 1, 1).ToArray();
+        Assert.AreEqual(3, result.Length);
+        Assert.IsFalse(result.Any(x => x.Key == trash.Key));
+        Assert.IsTrue(result[0].Key == children[0].Key);
+        Assert.IsTrue(result[1].Key == children[2].Key);
+        Assert.IsTrue(result[2].Key == children[3].Key);
+    }
+
+    [Test]
+    public void EntityService_Siblings_RespectsOrdering()
```

Comment:
I would be a lot happier if the order of the results would differ from the default sortorder. Because if the implementation gets changed where you remove the custom ordering, then this test will inform you everything is still fine.

=> pick 3 children en rename them to "X", "Y", "Z" then pick Y and check X and Z are before and after them